lib/collection/src/telemetry.rs

```diff
@@ -51,6 +52,20 @@ impl CollectionTelemetry {
             .map(|x| x.num_vectors.unwrap_or(0))
             .sum()
     }
+
+    /// Amount of optimizers currently running.
+    ///
+    /// Note: A `DetailsLevel` of 4 or setting `telemetry_detail.optimizer_logs` to true is required.
+    ///       Otherwise, this function will return 0, which may not be correct.
+    pub fn count_optimizers_running(&self) -> usize {
+        self.shards
+            .iter()
+            .flatten()
+            .filter_map(|replica_set| replica_set.local.as_ref())
+            .flat_map(|local_shard| local_shard.optimizations.log.iter().flatten())
+            .filter(|log| log.status == TrackerStatus::Optimizing)
+            .count()
+    }
```

Comment:
When reading the PR description I thought this could be done with counting with some atomic number and a guard.

But I think your approach is better, where we use state that we already have. 👍